src/xcode/ENA/ENA/Source/Scenes/RecycleBin/RecycleBinViewController.swift

```diff
@@ -0,0 +1,314 @@
+		}
+	}
+
+	// MARK: - Protocol UITableViewDataSource
+
+	override func numberOfSections(in tableView: UITableView) -> Int {
+		return viewModel.numberOfSections
+	}
+
+	override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+		return viewModel.numberOfRows(in: section)
+	}
+
+	override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+		switch RecycleBinViewModel.Section(rawValue: indexPath.section) {
+		case .description:
+			return descriptionCell(forRowAt: indexPath)
+		case .entries:
+			return recycleBinItemCell(forRowAt: indexPath)
+		case .none:
+			fatalError("Invalid section")
+		}
+	}
+
+	override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
+		viewModel.canEditRow(at: indexPath)
+	}
+
+	override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+		guard editingStyle == .delete else { return }
+
+		shouldReload = false
+		viewModel.removeEntry(at: indexPath)
+
+		tableView.performBatchUpdates({
+			var indexPaths = [indexPath]
+
+			if self.viewModel.isEmpty {
+				indexPaths.append(IndexPath(row: 0, section: RecycleBinViewModel.Section.description.rawValue))
+			}
+
+			tableView.deleteRows(at: indexPaths, with: .automatic)
+		}, completion: { _ in
+			self.shouldReload = true
+
+			if self.viewModel.isEmpty {
+				self.setEditing(false, animated: true)
+				self.updateEmptyState()
+			}
+		})
+	}
+
+	// MARK: - Protocol UITableViewDelegate
+
+	override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+		switch RecycleBinViewModel.Section(rawValue: indexPath.section) {
+		case .description:
+			break
+		case .entries:
+			showRestoreAlert(forItemAt: indexPath)
+		case .none:
+			fatalError("Invalid section")
+		}
+	}
+
+	override func tableView(_ tableView: UITableView, willBeginEditingRowAt indexPath: IndexPath) {
+		editModeEnteredBySwipeAction = true
+	}
+
+	// MARK: - Private
+
+	private let viewModel: RecycleBinViewModel
+
+	private var subscriptions = [AnyCancellable]()
+	private var shouldReload = true
+	private var editModeEnteredBySwipeAction = false
+
+	private func setupTableView() {
+		tableView.register(
+			DynamicTypeTableViewCell.self,
+			forCellReuseIdentifier: String(describing: DynamicTypeTableViewCell.self)
+		)
+
+		tableView.register(
+			UINib(nibName: String(describing: RecycleBinItemTableViewCell.self), bundle: nil),
+			forCellReuseIdentifier: String(describing: RecycleBinItemTableViewCell.self)
+		)
+
+		tableView.separatorStyle = .none
+		tableView.rowHeight = UITableView.automaticDimension
+		tableView.estimatedRowHeight = 60
+		tableView.accessibilityIdentifier = AccessibilityIdentifiers.TraceLocation.Overview.tableView
+	}
+
+	private func animateCellHeightChanges() {
+		DispatchQueue.main.async { [self] in
+			tableView.performBatchUpdates(nil, completion: nil)
```

Comment:
Why is this needed?

Reply:
Without this when entering the editing mode the cells retain their height even though their width is reduced by the delete icon that appears on the left side and in some cases that meant that labels were cut off. With this code the height is adjusted when entering and leaving editing mode.